### PR TITLE
Implement post dao

### PR DESCRIPTION
**Pull request by ethanliu24**

### CODE REVIEW

File: src/main/java/daos/DBPostDataAccessObject.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            Projections.include(field),
+            Projections.excludeId()
+        );
+        
+        Document doc = this.postRepository
+            .find(eq(field, target))
+            .projection(projectionFields)
+            .first();
+
+        return doc;
```
Comment (Karththigan) [inline src/main/java/daos/DBPostDataAccessObject.java:178]: So this code returns a Document that only contains the field `field` right?

E.g. if we searched for a post with entryID 1, the document returned would be {'entryID': 1}? Is this what is expected or do we want to return the entire Post itself (so all fields?)

File: src/main/java/daos/DBPostDataAccessObject.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            Projections.include(field),
+            Projections.excludeId()
+        );
+
+        // Retrieves documents that match the filter, applying a projection and a descending sort to the results
+        MongoCursor<Document> cursor = this.postRepository.find(lt(field, target))
+                .projection(projectionFields)
+                .sort(Sorts.descending(POSTED_DATE)).iterator();
+        
+        return cursor;
```
Comment (Karththigan) [inline src/main/java/daos/DBPostDataAccessObject.java:197]: Same question as above, do we want to return all fields in each post Document? Or just the specific field?

File: src/main/java/daos/DBPostDataAccessObject.java (hunk 1 above)
Comment (ethanliu24) [reply src/main/java/daos/DBPostDataAccessObject.java:178]: All fields. Document is a BSON object (which is similar to json). We are query a post with the given field, e.g. a given post id. A BSON document is returned and in the overriden functions they could decide what to do with it

File: src/main/java/daos/DBPostDataAccessObject.java (hunk 2 above)
Comment (ethanliu24) [reply src/main/java/daos/DBPostDataAccessObject.java:197]: We return all fields (all "attributes") of a post. For example, if we look for a post in category `general` its gonna return all posts with the `general` value in the `category` field

File: src/main/java/daos/DBPostDataAccessObject.java (hunk 1 above)
Comment (Karththigan) [reply src/main/java/daos/DBPostDataAccessObject.java:178]: So line 169 does not only include the field that was inputted?

Comment (ethanliu24) [reply src/main/java/daos/DBPostDataAccessObject.java:178]: no, the field is the thing we search and match for in the database

Comment (Karththigan) [reply src/main/java/daos/DBPostDataAccessObject.java:178]: Ok!

**Review by Karththigan — APPROVED** [review]

**Pull request merged**